### DISCUSSION TITLE
Matching 3 package files but only extracting 1 package file

### DISCUSSION THREAD

**Discussion by linzh92** [renovatebot/renovate#22764] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Matching 3 package files and extracting 3 package file. But I have got such result as below: 
<img width="645" alt="image" src="https://github.com/renovatebot/renovate/assets/37093133/570b61dd-2851-43ec-a213-395eafbd17d1">

### What you tried so far.

View the source code of renovate.
<img width="1001" alt="image" src="https://github.com/renovatebot/renovate/assets/37093133/805bba02-a02a-4e55-bfd6-6a106cd6c075">

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
This means it matched 3 files based on file patterns, but 2 of 3 of them don't have dependencies detected inside
